### PR TITLE
Drop Python 3.6

### PR DESCRIPTION
**Pull request by joshmoore**
see: #317
see: https://github.com/zarr-developers/zarr-python/issues/765

- [ ] Unit tests and/or doctests in docstrings
- [ ] `tox -e py39` passes locally
- [ ] Docstrings and API docs for any new/modified user-facing classes and functions
- [ ] Changes documented in docs/release.rst
- [ ] `tox -e docs` passes locally
- [ ] GitHub Actions CI passes
- [ ] Test coverage to 100% (Coveralls passes)

### CODE REVIEW

**Comment by joshmoore** [comment]
@jakirkham, thoughts or help on the feedstock front(s) welcome.

File: setup.py [hunk 1]
```diff
@@ -346,7 +346,7 @@ def run_setup(with_extensions):
         ext_modules=ext_modules,
         cmdclass=cmdclass,
         package_dir={"": "."},
-        python_requires=">=3.6, <4",
+        python_requires=">=3.7, <4",
```
Comment (joshmoore) [inline setup.py:349]: This change is likely debatable since we're not explicitly breaking compatibility.

**Comment by jakirkham** [comment]
Might be worth updating `cibuildwheel`. Also think we can skip certain builds. From [their docs]( https://cibuildwheel.readthedocs.io/en/stable/options/#examples_1 ), this looks useful:

```yaml
# Skip building on CPython 3.6 on all platforms
CIBW_SKIP: cp36-*
```

File: setup.py (hunk 1 above)
Comment (jakirkham) [reply setup.py:349]: Eh I think it is ok to drop. We are not testing it any more. So we really can't guarantee things work there.

**Comment by joshmoore** [comment]
Changed 3.6 branch protection for `master` to 3.10. Closing and re-opening.

**Comment by joshmoore** [comment]
The hanging `build (3.6) Expected — Waiting for status to be reported` build apparently isn't going away, but the build will pass without.

**Comment by jakirkham** [comment]
May need to push another commit

**Comment by joshmoore** [comment]
Good try, but I think it takes a new PR 😉

**Comment by jakirkham** [comment]
Yeah it may. Sorry pushed a couple more things. There was an old notebook that I reran. It has an issue we might need to follow up on ( https://github.com/zarr-developers/numcodecs/issues/319 ). Also added a release note. Otherwise LGTM

Thanks for working on this Josh! 😄 🙏

**Comment by jakirkham** [comment]
Hmm..was hoping that would trigger RTD without having to restart all the builds. Doesn't seem to have, but won't push on that any further in this PR.

**Comment by jakirkham** [comment]
So the wheel job has actually completed, but the status is still showing pending. Going to merge anyways.

**Comment by jakirkham** [comment]
Still seeing the Python 3.6 status as pending in PR ( https://github.com/zarr-developers/numcodecs/pull/320 ). Guessing we are missing something

**Comment by jakirkham** [comment]
Think I figured it out. Python 3.6 was still listed in the branch protection rules for `master` on GitHub. Removing it from that list seems to have fixed the issue.

**Pull request merged**